src/components/Icon.tsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+
+import Arrow from "@assets/icons/arrow.svg";
+import Back from "@assets/icons/back.svg";
+import Calendar from "@assets/icons/calendar.svg";
+import Check from "@assets/icons/check.svg";
+import Down from "@assets/icons/down.svg";
+import Edit from "@assets/icons/edit.svg";
+import Info from "@assets/icons/info.svg";
+import List from "@assets/icons/list.svg";
+import Location from "@assets/icons/location.svg";
+import Plus from "@assets/icons/plus.svg";
+import Profile from "@assets/icons/profile.svg";
+import Temperature from "@assets/icons/temperature.svg";
+import Trash from "@assets/icons/trash.svg";
+import DownWhite from "@assets/icons/down_white.svg";
+import TemperatureGray from "@assets/icons/temperature_gray.svg";
+import { Colors } from "@/src/styles/Colors";
+import { ColorValue } from "react-native";
+import Svg, { Mask, Rect } from "react-native-svg";
+
+export enum IconName {
+  arrow = "arrow",
+  back = "back",
+  calendar = "calendar",
+  check = "check",
+  down = "down",
+  edit = "edit",
+  info = "info",
+  list = "list",
+  location = "location",
+  plus = "plus",
+  profile = "profile",
+  temperature = "temperature",
+  trash = "trash",
+  downWhite = "down_white",
+  temperatureGray = "temperature_gray",
+}
+
+const icons = {
+  [IconName.arrow]: Arrow,
+  [IconName.back]: Back,
+  [IconName.calendar]: Calendar,
+  [IconName.check]: Check,
+  [IconName.down]: Down,
+  [IconName.edit]: Edit,
+  [IconName.info]: Info,
+  [IconName.list]: List,
+  [IconName.location]: Location,
+  [IconName.plus]: Plus,
+  [IconName.profile]: Profile,
+  [IconName.temperature]: Temperature,
+  [IconName.trash]: Trash,
+  [IconName.downWhite]: DownWhite,
+  [IconName.temperatureGray]: TemperatureGray,
+};
+
+interface IconProps {
+  name: IconName;
+  size?: number;
+}
+
+const Icon = ({ name, size = 24 }: IconProps) => {
+  const SvgIcon = icons[name];
+  return <SvgIcon width={size} height={size} />;
+};
```

Comment:
여기 부분에서는 아래처럼 진행하려고 했었는데, SvgIcon의 Property 에서는 color 가 잘 작동하지 않았습니다. 

```suggestion
const Icon = ({ name, size = 24, color = Colors.black40 }: IconProps) => {
  const SvgIcon = icons[name];
  return <SvgIcon width={size} height={size} color={color}/>;
};
```

개선 방향으로 masking 을 활용해 색상 변경을 처리할 수 있는 방법을 찾아보려 했는데, 추가적인 패키지 설치나 복잡한 유틸 코드가 포함되어야 하는 것으로 보여서, 간편하게 수정할 수 있는 방법이 있을지 논의해보고자 하였습니다